### PR TITLE
Check the docker daemon MTU setting

### PR DESCRIPTION
**Pull request by niusmallnan**
https://github.com/rancher/rancher/issues/7583

### CODE REVIEW

**Comment by niusmallnan** [comment]
if the Docker daemon is configured with different MTU than the configured when launching the catalog. 
```
..........
time="2017-08-28T13:50:29Z" level=error msg="Failed to apply cni conf: Get different MTU, cni:1600, docker:1450"
..........
```

File: cniconf/watcher.go [hunk 1]
```diff
@@ -90,6 +92,13 @@ func (w *watcher) apply(network metadata.Network, host metadata.Host) error {
 	var lastErr error
 	for file, config := range cniConf {
 		config = utils.UpdateCNIConfigByKeywords(config, host)
+
+		err := checkMTU(config)
+		if err != nil {
+			lastErr = err
+			continue
```
Comment (leodotcloud) [inline cniconf/watcher.go]: Don't `continue`, just a log is good.
```
logrus.Errorf("cniconf: Docker Bridge MTU %v is different from CNI config MTU: %v", dockerBridgeMTU, cniConfigMTU);
```

File: cniconf/watcher.go [hunk 2]
```diff
@@ -128,3 +137,27 @@ func (w *watcher) apply(network metadata.Network, host metadata.Host) error {
...
+	c, err := events.NewDockerClient()
+	if err != nil {
+		return err
+	}
+
+	networks, err := c.ListNetworks()
+	if err != nil {
+		return err
+	}
+	for _, n := range networks {
```
Comment (leodotcloud) [inline cniconf/watcher.go]: Can you please split the functionality of getting the MTU from docker as a separate function?
`func GetDockerNetworkBridgeMTU(string bridgeName)`

**Review by leodotcloud — CHANGES_REQUESTED** [review]

**Comment by leodotcloud** [comment]
@niusmallnan do the changes work if we change the bridge in IPSec or VXLAN catalog from docker0 to rancher0?

File: cniconf/watcher.go [hunk 3]
```diff
@@ -128,3 +132,37 @@ func (w *watcher) apply(network metadata.Network, host metadata.Host) error {
...
 }
+
+func checkMTU(config interface{}) {
+	props, _ := config.(map[string]interface{})
+	bridgeName, _ := props["bridge"].(string)
+	cniConfigMTU := fmt.Sprintf("%.0f", props["mtu"])
+
+	dockerBridgeMTU, exist, err := getDockerNetworkBridgeMTU(bridgeName)
+	if err != nil {
+		logrus.Errorf("checkMTU: Get error from docker api: %s", err)
```
Comment (leodotcloud) [inline cniconf/watcher.go]: Minor typo: Should be `Got error from docker api:`

**Comment by niusmallnan** [comment]
@leodotcloud 
In that case, rancher0 bridge will be created by `rancher-cni-bridge` and get the correct MTU.
In addition, these code will determine whether the bridge is Docker Bridge.

**Review by leodotcloud — APPROVED** [review]

**Comment by leodotcloud** [comment]
LGTM

**Pull request merged**